File: Code/05_File_POSIX/Ex8.c
#include <stdio.h>
#include <fcntl.h>
#include <unistd.h>
#include <stdlib.h>

int main(int argc, const char * argv[])
{
	int fp;
	char str[100] = {'\0'};
	float pi = 3.1415;
	float pilido;

	fp = open ("exercicio8.txt",
		O_RDWR | O_CREAT, S_IRWXU);
	if(fp==-1)
	{
		printf("Erro na abertura"
			" do arquivo.\n");
		exit(1);
	}
	sprintf(str, "pi = %f\n", pi);
	write(fp, str, 100);
	close(fp);

	fp = open("exercicio8.txt",
		O_RDONLY);
	read(fp, str, 100);
	sscanf(str, "pi = %f\n", &pilido);
	close(fp);

	printf("\nO valor de PI (lido"
		" do arquivo) eh: %f\n\n",
		 pilido);
	return(0);
}

File: 2_Programação_sistemas_operacionais/2.2_File_POSIX/Python/Ex8.py
import os, struct, stat

try:
	fd = os.open("exercicio8.txt", os.O_RDWR | os.O_CREAT, stat.S_IRWXU)
except OSError:
	print("Erro na abertura do arquivo.")
	exit(-1)
pi = 3.1415
os.write(fd, str.encode("pi = %f" % pi))
os.close(fd)
fd = os.open("exercicio8.txt", os.O_RDONLY)
cnt = ''
c = os.read(fd, 100)
while len(c):
	cnt += c.decode("utf-8")
	c = os.read(fd, 100)
c1,c2,pilido = cnt.split()
print("O valor de PI (lido do arquivo) é", pilido)
os.close(fd)